blocks/editable/patterns.js

```diff
@@ -219,6 +221,37 @@ export default function( editor ) {
 		onReplace( [ block ] );
 	}
 
+	function character() {
+		if ( ! onReplace ) {
+			return;
+		}
+
+		// Merge text nodes.
+		editor.getBody().normalize();
```

Comment:
How expensive an operation is this for something which will occur on every single keypress? _Edit:_ I guess it's every keypress for a block which implements `onReplace` (inserter block).